### DISCUSSION TITLE
firmware 23.44.0 any way to get access to neurio data?

### DISCUSSION THREAD

**Discussion by anderep** [jasonacox/Powerwall-Dashboard#424] [General]
first i need to start out with a big thank you to Jasonacox for all your hard work on Powerwall_Dashboard.

well i finally got pushed the firmware (down) upgrade, sad day.  i was mining data from the (solar) neurio for both solar systems, one CT for Tesla inverter and another CT for Enphase.

i want to get access to this data again, any hope with new firmware?

as much as i don't want to, I can put the neurio's on the local network and mine the data, perhaps?  any ideas?

Or probably should just go by a product like Sence, Emporia or something else that i can configure to have two solar circuits. any ideas?

**Comment by jasonacox** [comment]
Hi @anderep -

The Neurios that are installed by Tesla are built with a propriety firmware for Tesla. It is very unlikely you will be able to modify them. During some troubleshooting of mine, I discovered that they were talking to the local Powerwall WiFi access point.  I sent a note to the Generac (new owner of Neurio) to see if there was a way to access them, and received this:

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/85eac009-f9da-48bb-b5a0-a4026f3ca915)

I'm not yet on 23.44.0, but like you, I will be looking to alternatives to get circuit and other data. It will be interesting to see what others in the community find.

Long term, I'm hopeful. I think it makes sense to keep a local API on the gateway, for no other reason than to provide owners with the ability to monitor/troubleshoot their system during power and internet outage events. Hopefully that will include some of the extended data we have been getting from device vitals.

**Comment by anderep** [comment]
When you power cycle the neurio, it hosts an access point for a few minutes where you can login and connect to a network.  The teg will scan the local network for devices and connect.

**Reply by jasonacox** [reply]
Interesting, thanks @anderep . When I had trouble with mine which is built in to the Powerwall+, I could connect but never could get it to do anything more. It turned out to be a bad Neurio and was replaced. I haven't tried since.

**Reply by cwagz** [reply]
@anderep - Is there a benefit to adding the Neurio devices to the local network other than potentially a better WiFi connection?  My garage access point is straight through the exterior wall from the Neurio’s so the signal should be much stronger than from the TEG which ends up being at like 10 degrees through the exterior wall. My understanding is that this makes the wall much harder to reliably pass signals through.

**Reply by anderep** [reply]
Perhaps ability to mine data directly from neurio…

there are instructions on tesla motor energy forum.

**Reply by cwagz** [reply]
For my system it looks like I would need to delete the meters and then add them as new. I ended up reverting back to direct WiFi to TEG. I see where the MAC and IP can be entered but there does not appear to be a way to edit the existing meter connections in the TEG.

**Reply by finasfinas** [reply]
@anderep 
Can you link me to the instructions to mine data directly from my tesla neurio? I want to get the energy the house is consuming but I can only see that using the tesla one app. My neurio is connected to a gen3 wall charger. 

I was able to connect the neurio to my wifi network, it has port 443 open but it does not return anything.

many thanks.

**Reply by anderep** [reply]
I mine the data from my neurio from the teg using powerwall dashboard.

**Reply by finasfinas** [reply]
that does not work for me as I don't have a powerwall, I have a gen3 wall charger.
